{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sample 2: Hail Filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Atgenomix-Sample-Hail-Filter\n",
    "# This demo shows \n",
    "# 1. how to use atgenomix-hail to filter variants based on different requirements \n",
    "# 2. its corresponding performance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pyspark import SparkContext\n",
    "from pyspark.sql import SQLContext\n",
    "from pyspark.sql.functions import struct\n",
    "from pyspark.sql.types import *\n",
    "import hail as hl\n",
    "import datetime\n",
    "import time\n",
    "\n",
    "sc = SparkContext.getOrCreate()\n",
    "sc.setLogLevel(\"ERROR\")\n",
    "hl.init(sc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Demo Purpose\n",
    "# Load a Matrix Table generated by annotating (clinvar, dbNSFP, gencode, gnomad) to a whole genome VCF\n",
    "# This Matrix Table file has 4,941,747 variants.\n",
    "\n",
    "mt = hl.atgenomix.load_sample_mt()\n",
    "print(mt.count())\n",
    "mt.describe()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
